### PR TITLE
Feature/design dashboard

### PR DESCRIPTION
**Pull request by lalonggone**
# PULL REQUEST

## 🚀 Changes

- Added ... Total $ spent and the traveler's past, upcoming and pending trips. Added very little styling.. definitely need to dig into that more later. Looks like 💩 right now !
- Refactored ... I separated some functionality from the getAllData promise in scripts.js to make it easier to read. I also refactored some of the structure in that promise, again for better organization and clarity. The HTML is a little different since I'm passing data into the trip-type containers dynamically.

## 📸 Screenshots
Screenshot 2024-03-02 at 5 08 17 PM

## 📝 Additional Notes

the aside is pictured above.. it is currently commented out in the code. I'm toying with the idea of making the new trip form with the aside or as a section that is centered / inline with the trip containers. 

I also want to make the trip cards into a grid but was struggling on it for too long.. come back later

I am aware that I have no tests yet... 😅

## ✅ Checklist before merge:

Make sure you've completed the following steps before merging this PR:
- [x] I have performed a self-review of my own code.
- [x] Changes and updates are well-documented in this PR and/or the README. Any known issues are in the notes section, otherwise there are none.
- [x] I double checked the PR in the preview and took my time writing the PR.

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -7,18 +7,34 @@
...
+    <section class="header-container">
+      <h1 class="header-title">TRAVEL TRACKER</h1>
+      <section class="header-right">
+        <h2 class="welcome-title">Welcome, User</h2>
+        <h3 class="total-spent">You've spent $$$</h2>
+      </section>
+    </section>
+  </header>
+  <main>
+    <!-- <aside class="aside">
```
Comment (tednaphil) [inline dist/index.html:20]: I like your idea of having the place to plan a new trip in a very visible aside! I think that'll make it easier on the user to plan in fewer clicks

File: src/apiCalls.js [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-            console.log(singleTraveler, allTrips, allDestinations)
-        })
-        .catch(error => console.log(error))
+function getAllData() {
+    return Promise.all([
+        fetch('http://localhost:3001/api/v1/travelers').then(resp => resp.json()),
+        fetch('http://localhost:3001/api/v1/trips').then(resp => resp.json()),
+        fetch('http://localhost:3001/api/v1/destinations').then(resp => resp.json())
+    ])
+    .then(([travelers, trips, destinations]) => {
```
Comment (tednaphil) [inline src/apiCalls.js:10]: fabulous idea returning all the data in an object for easy and organized access in scripts!

When you move to error handling, checking if the response's `ok` property is `true` and then moving on to parsing might take a little refactoring of lines 6-8, but might be a good place for the safety net that likely won't get used at all

File: src/scripts.js [hunk 3]
```diff
@@ -1,13 +1,76 @@
...
+        });
+
+        displayTotalCost(totalCost)
+        checkAndDisplayEmptyMessage('.pending-trips-container', 'pending');
+        checkAndDisplayEmptyMessage('.past-trips-container', 'past');
+        checkAndDisplayEmptyMessage('.approved-trips-container', 'upcoming');
+    });
+});
+
+function processTrips(travelerTrips, destinations) {
```
Comment (tednaphil) [inline src/scripts.js:46]: I love that all the trip's information is put together here in one object

File: src/scripts.js [hunk 4]
```diff
@@ -1,13 +1,76 @@
...
+
+function calculateTotalTripCost(travelerTrips, destinations) {
+    let totalCost = 0;
+
+    travelerTrips.forEach(trip => {
+        const destination = destinations.find(dest => dest.id === trip.destinationID);
+
+            const tripFlightCost = destination.estimatedFlightCostPerPerson * trip.travelers;
+            const tripLodgingCost = destination.estimatedLodgingCostPerDay * trip.duration;
+            // console.log(tripFlightCost, tripLodgingCost);
```
Comment (tednaphil) [inline src/scripts.js:69]: Will the travel agent's fee be calculated here also, or in a separate function?

**Review by tednaphil — APPROVED** [review]
I love that you're using SASS! I'm really excited to see your ideas come to life when you get to styling elements. 

I think the way you have your data manipulation setup should make accessing information about trips so much easier. Also, great work using semantic tags!

File: src/scripts.js (hunk 4 above)
Comment (lalonggone) [reply src/scripts.js:69]: thanks for this reminder!! I think it makes sense to have it in the calculate trip costs function

**Pull request merged**